jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -800,23 +808,41 @@ def _register_spectrum(self, event):
         self.add_results.add_results_from_plugin(spectrum)
         self._set_default_results_label()
 
-    def _apply_subset_mask(self, spectrum):
-        """"
-        Apply mask from selected subset to ``spectrum``
+    def _apply_subset_masks(self, spectrum, dimensions=['spectral', 'spatial']):
         """
-        if self.spectral_subset_selected != "Entire Spectrum":
-            subset_mask = self.app.get_data_from_viewer(
-                self._default_spectrum_viewer_reference_name,
-                data_label=self.spectral_subset_selected
-            ).mask
+        Iterate over dimensions of a spectral cube and mask out the
+        subsets along each axis
+        """
+        for dimension in dimensions:
+            if dimension == 'spectral' and hasattr(self, '_default_spectrum_viewer_reference_name'):
+                viewer_ref = self._default_spectrum_viewer_reference_name
+                data_label = self.spectral_subset_selected
+            elif dimension == 'spatial' and hasattr(self, '_default_flux_viewer_reference_name'):
+                viewer_ref = self._default_flux_viewer_reference_name
+                data_label = self.spatial_subset_selected
+            else:
+                # If this viewer isn't available, don't try to apply the subset mask
+                continue
 
-            if subset_mask.shape != spectrum.shape:
-                # broadcast to the dimensions of the full cube:
-                subset_mask = np.broadcast_to(
-                    subset_mask[None, None, :], spectrum.shape
+            if (not getattr(self, dimension + '_subset_selected').startswith("Entire") and
+                    viewer_ref in self.app.get_viewer_reference_names()):
+                subset = self.app.get_data_from_viewer(
+                    viewer_ref, data_label=data_label
                 )
+                if hasattr(subset, 'mask'):
+                    # the mask attr is available for spectral subsets:
+                    subset_mask = subset.mask
+                else:
+                    # otherwise, `subset` is a GroupedSubset:
+                    subset_mask = subset.to_mask()
 
-            if spectrum.mask is None:
-                spectrum.mask = subset_mask
-            else:
-                spectrum.mask += subset_mask
+                if subset_mask.shape != spectrum.shape:
+                    # broadcast to the dimensions of the full cube:
+                    subset_mask = np.broadcast_to(
+                        subset_mask[None, None, :], spectrum.shape
+                    )
+
+                if spectrum.mask is None:
+                    spectrum.mask = subset_mask
+                else:
+                    spectrum.mask += subset_mask
```

Comment:
any downside to adding (vs multiplying)?  I see this is how it was previously as well, but now that we're combining spatial and spectral, this code might actually get triggered whereas it probably wasn't before unless a mask already existed on the input data.

Reply:
I agree that addition isn't right here. I think what we actually want is 
```python
spectrum.mask |= subset_mask
```
If either the spectral or the spatial coordinate has `mask==True`, we want to exclude that elements from the fit. We don't want `*` or `&` because that would only exclude cube elements where the mask is `True` for both the spectral and spatial masks.

Reply:
I believe the masks are actually integers, not booleans, hence why addition works here. Anything other than 0 is masked.

Reply:
Just checked: the output of `get_subset_object` is an `ndarray` with `dtype('bool')`.

Reply:
I see, not what I expected. Checking the specutils docs it does allow either boolean or integer `mask` though, I remembered incorrectly that they're always integer.

Reply:
For booleans, there probably isn't any difference between `|=` and `+=`, but I think it's more readable to change (or add a comment reminding that they're booleans).